src/test/groovy/com/mateusz/jakuszko/roomforyou/service/ApartmentDbServiceSpecification.groovy

```diff
@@ -0,0 +1,142 @@
+package com.mateusz.jakuszko.roomforyou.service
+
+import com.mateusz.jakuszko.roomforyou.entity.Apartment
+import com.mateusz.jakuszko.roomforyou.entity.Reservation
+import com.mateusz.jakuszko.roomforyou.exceptions.NotFoundException
+import com.mateusz.jakuszko.roomforyou.repository.ApartmentRepository
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import spock.lang.Specification
+
+@SpringBootTest
+class ApartmentDbServiceSpecification extends Specification {
+
+    @Autowired
+    private ApartmentDbService service
+    @Autowired
+    ApartmentRepository repository
+
+    def "should throw an exception when there isn't such apartment"() {
+        when: "pull apartment from database which doesn't exist"
+        Optional<Apartment> apartment = service.getApartment(123)
+        apartment.orElseThrow(NotFoundException::new)
+
+        then:
+        thrown(NotFoundException)
```

Comment:
Ten wyjątek nie jest potrzebny. Proponuje w ten sposób:
```
        when: 
        Optional<Apartment> apartment = service.getApartment(123)

        then:
        apartment.isEmpty()
```

Reply:
No i tak naprawdę tytuł powinien brzmieć "Should not fetch any apartament for nonexistent id" albo coś tym stylu bo ten wyjątek nie jest przecież rzucany przez serwis, a testy dla danego komponentu powinniśmy postrzegać jak zbiór wymagań wobec niego (i przy okazji jego dokumentacja)

Reply:
Dzięki za cenne uwagi!